src/DS/sds.c

```diff
@@ -255,16 +266,11 @@ int ds_sds_dump_component_ref_as(xmlNodePtr component_ref, struct ds_sds_session
 	char* filename_cpy = oscap_sprintf("./%s", relative_filepath);
 	char* file_reldir = dirname(filename_cpy);
 
-	// the cast is safe to do because we are using the GNU basename, it doesn't
-	// modify the string
-	const char* file_basename = basename((char*)relative_filepath);
-
-	const char* target_filename_dirname = oscap_sprintf("%s/%s", target_dir, file_reldir);
-	const char* sce_filename = oscap_sprintf("%s/%s/%s", target_dir, file_reldir, file_basename);
-	ds_sds_dump_component(component_id, session, sce_filename, relative_filepath);
-	oscap_free(sce_filename);
+	const char* target_filename_dirname = oscap_sprintf("%s/%s",sub_dir, file_reldir);
 	oscap_free(filename_cpy);
 
+	ds_sds_dump_component(component_id, session, target_filename_dirname, relative_filepath);
+
```

Comment:
I think that there are some needless operations.
* `filename_cpy` is always `./relative_filepath` - in other words it always starts with `./`
* therefore calling `dirname(filename_cpy)` will always return `.`
* then `file_reldir` will always be `.`
* then `target_filename_dirname` will be `sub_dir/.`. I find it weird that this path will always finish with `.`
* Conslusion: `target_filename_dirname` equals `sub_dir` (from the point of view of unix paths)

I think you can remove lines 266-271 and call the function `ds_sds_dump_component` like this: 
```
ds_sds_dump_component(component_id, session, sub_dir, relative_filepath)
```

The variable `target_filename_dirname` is used also in recursive call of `ds_sds_dump_component_ref_as()` at line 327. I think we can also use `sub_dir` there instead of `target_filename_dirname`.